src/gc.c

```diff
@@ -1556,6 +1577,47 @@ static void gc_pool_sync_nfree(jl_gc_pagemeta_t *pg, jl_taggedvalue_t *last) JL_
     pg->nfree = nfree;
 }
 
+void gc_sweep_pool_parallel(jl_ptls_t ptls, int master) JL_NOTSAFEPOINT
+{
+    size_t lb;
+    size_t ub;
+    size_t n_pckts = gc_sweep_queue.current - gc_sweep_queue.begin;
+    if (master) {
+        lb = 0;
+        ub = n_pckts / (jl_n_gcthreads + 1);
+    }
+    else {
+        lb = ((ptls->tid - gc_first_tid + 1) * n_pckts) / (jl_n_gcthreads + 1);
+        ub = ((ptls->tid - gc_first_tid + 2) * n_pckts) / (jl_n_gcthreads + 1);
+    }
+    for (gc_sweep_packet_t *pp = gc_sweep_queue.begin + lb; pp < gc_sweep_queue.begin + ub; pp++) {
+        sweep_packet(pp);
+    }
+}
+
+static void gc_sweep_pool_concurrent(void)
```

Comment:
Add some docstrings explaining the structure of the code, and why you settled on this tradeoff.